### DISCUSSION TITLE
Question about dataset

### DISCUSSION THREAD

**Discussion by hjkopaowjsk** [lenskit/lkpy#314] [Q&A]
I want to use my own dataset which is not all numbers. Is it possible? (I got an error while testing.)

**Comment by mdekstrand** [comment]
What do you mean by "not all numbers"? Are you looking to use alphanumeric user and/or item IDs?

Doing so is possible, but often inefficient.

Could you share a small sample of your data set and the error message you are experiencing? That will help with debugging.

**Reply by hjkopaowjsk** [reply]
I did an extra try and now they are all numbers. But there is another error, the list is empty.
![error1](https://user-images.githubusercontent.com/104627786/166098103-79ccedba-cbff-43c8-84ae-063a8266cd75.png)

In addition, the ratings in my dataset are only 0 and 1. Will this affect his work? Do you have any suggestions for this?
![dataset](https://user-images.githubusercontent.com/104627786/166088386-d0545f46-48cc-49a3-95a9-ff522eb7da36.png)

**Reply by mdekstrand** [reply]
Ratings of 0 and 1 can work, but usually need some more finesse. Do you actually have any 0s, are are all observed ratings 1?

Do you have a notebook you could share? It looks like you are probably using the `eval` function pretty similar to the example docs, but without more details on what's happening with the data, and any modifications you made to the `eval` function, it's hard to know what is happening here.
